jumanji/environments/logic/sliding_tile_puzzle/env_test.py

```diff
@@ -0,0 +1,155 @@
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import chex
+import jax
+import jax.numpy as jnp
+
+from jumanji.environments.logic.sliding_tile_puzzle import SlidingTilePuzzle
+from jumanji.environments.logic.sliding_tile_puzzle.types import State
+from jumanji.testing.env_not_smoke import check_env_does_not_smoke
+from jumanji.testing.pytrees import assert_is_jax_array_tree
+from jumanji.types import TimeStep
+
+
+def test_sliding_tile_puzzle_reset_jit(sliding_tile_puzzle: SlidingTilePuzzle) -> None:
+    """Confirm that the reset method is only compiled once when jitted."""
+    chex.clear_trace_counter()
+    reset_fn = jax.jit(chex.assert_max_traces(sliding_tile_puzzle.reset, n=1))
+    key = jax.random.PRNGKey(0)
+    state, timestep = reset_fn(key)
+
+    # Verify the data type of the output.
+    assert isinstance(timestep, TimeStep)
+    assert isinstance(state, State)
+
+    # Check that the state is made of DeviceArrays, this is false for the non-jitted.
+    assert_is_jax_array_tree(state.puzzle)
+    assert_is_jax_array_tree(state.empty_tile_position)
+
+    # Call again to check it does not compile twice.
+    state, timestep = reset_fn(key)
+    assert isinstance(timestep, TimeStep)
+    assert isinstance(state, State)
+
+
+def test_sliding_tile_puzzle_step_jit(sliding_tile_puzzle: SlidingTilePuzzle) -> None:
+    """Confirm that the step is only compiled once when jitted."""
+    key = jax.random.PRNGKey(0)
+    state, timestep = jax.jit(sliding_tile_puzzle.reset)(key)
+    action = jnp.array(0)
+
+    chex.clear_trace_counter()
+    step_fn = jax.jit(chex.assert_max_traces(sliding_tile_puzzle.step, n=1))
+
+    new_state, next_timestep = step_fn(state, action)
+
+    # Check that the state has changed.
+    assert not jnp.array_equal(new_state.puzzle, state.puzzle)
+
+    # Check that the state is made of DeviceArrays, this is false for the non-jitted.
+    assert_is_jax_array_tree(new_state)
+
+    # New step
+    state = new_state
+    new_state, next_timestep = step_fn(state, action)
+
+    # Check that the state has changed
+    assert not jnp.array_equal(new_state.puzzle, state.puzzle)
+
+
+def test_sliding_tile_puzzle_get_action_mask(
+    sliding_tile_puzzle: SlidingTilePuzzle,
+) -> None:
+    """Verify that the action mask generated by `_get_valid_actions` is correct."""
+    key = jax.random.PRNGKey(0)
+    state, _ = sliding_tile_puzzle.reset(key)
+    get_valid_actions_fn = jax.jit(sliding_tile_puzzle._get_valid_actions)
+    action_mask = get_valid_actions_fn(state.empty_tile_position)
+
+    # Check that the action mask is a boolean array with the correct shape.
+    assert action_mask.dtype == jnp.bool_
+    assert action_mask.shape == (4,)
+
```

Comment:
I know this is a bit of a mission but can you double check here that when the empty position is on the edge the mask works correctly?

Can just use the setup from `test_env_illegal_move_does_not_change_board` and check the action mask